app/models/manageiq/providers/microsoft/infra_manager/vm_or_template_shared/scanning.rb

```diff
@@ -0,0 +1,79 @@
+module ManageIQ::Providers::Microsoft::InfraManager::VmOrTemplateShared::Scanning
+  def perform_metadata_scan(ost)
+    require 'MiqVm/miq_scvmm_vm'
+
+    log_pref = "MIQ(#{self.class.name}##{__method__})"
+    $log.debug "#{log_pref} VM = #{name}"
+
+    begin
+      setup_for_ems_connection(name, ost)
+      miq_vm = MiqScvmmVm.new(name, ost)
+      scan_via_miq_vm(miq_vm, ost)
+    rescue => err
+      $log.error "#{log_pref}: #{err}"
+      $log.debug err.backtrace.join("\n")
+      raise
+    ensure
+      miq_vm.unmount if miq_vm
+    end
+  end
+
+  def perform_metadata_sync(ost)
+    sync_stashed_metadata(ost)
+  end
+
+  def validate_smartstate_analysis
+    validate_supported_check("Smartstate Analysis")
+  end
+
+  def scan_via_ems?
+    false
+  end
+
+  private
+
+  # Moved from MIQExtract.rb
+  # TODO: Should this be in the ems?
+  def setup_for_ems_connection(vm_name, ost)
```

Comment:
Yeah I don't know why I wasn't getting what you were saying but that makes perfect sense.  I'll make the required changes here and in miq_scvmm_vm.